pkg/pdatautil/hash_test.go

```diff
@@ -136,6 +136,26 @@ func TestMapHash(t *testing.T) {
 	}
 }
 
+// TestMapHashPanic tests a very specific case where a panic can occur with nested maps.
+// Outlined in https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/18910
+func TestMapHashPanic(t *testing.T) {
```

Comment:
Can this be converted into a case in the above test? The test won't pass if it panics, right? I think the comment w/ link is useful for context, but it can be noted along with the case.